index/scorch/snapshot_index.go

```diff
@@ -51,6 +51,15 @@ type asynchSegmentResult struct {
 
 var reflectStaticSizeIndexSnapshot int
 
+// DefaultFieldTFRCacheThreshold limits the number of TermFieldReaders(TFR) for
+// a field in an index snapshot. Without this limit, when recycling TFRs, it is
+// possible that a very large number of TFRs may be added to the recycle
+// cache, which could eventually lead to significant memory consumption.
+// This threshold can be overwritten by users at the library level by changing the
+// exported variable, or at the index level by setting the FieldTFRCacheThreshold
+// in the kvConfig.
+var DefaultFieldTFRCacheThreshold uint64 = 10
```

Comment:
There a reason why you chose 10?

Reply:
Not really, I just picked it arbitrarily 😬 
I'll consider choosing another value with some deliberation, thanks!

Reply:
I don't think there is a universal perfect number here. And once the applications realize such a magic number that works for them through trials then they could configure it either index level or library level depending on their use case.
Have the same thoughts for cbft too.